packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -87,6 +87,10 @@ async function setupResourceServerSession(
 ): Promise<void> {
   const webIdAsUrl = new URL(webId);
   const resourceServerIri = webIdAsUrl.origin;
+  // Querying the /session endpoint does not set the cookie, but issuing an
+  // authenticated request to any actual resource (even public ones) does,
+  // so we fetch the user's WebID before checking the /session endpoint.
+  await authenticatedFetch(webId);
```

Comment:
In the code just below this, after we check the RS gives us back a 200, shouldn't we be checking that the value of the WebID in the response body also matches what we think our WebID is (i.e. the value of the passed-in `webId` parameter)...?

Reply:
Technically we could, but I'm not sure how much security it would add: the cookie would still be set regardless, so the user would have access to the resource server based on the WebID of the cookie, even in the (unlikely) case where it doesn't match the WebID attached to the ID token and to the Access token. I only say it's unlikely because only ESS will ever implement this cookie-based session management, and given it's a temporary workaround, I don't think it creates much value to consider edge cases, but rather we should move quick to a cleaner solution. What do you think ?

Reply:
Agreed that it would be better to do that if this were a long-term solution, but given the temporary nature of this and the likely usages, I also agree that there's no need to implement this now. (And to avoid the risk of there being nothing as permanent as a temporary solution, it might be _better_ not to treat it as something permanent.)

Reply:
Sure, I get the short-term situation thing. As a minimal-work compromise then, I'd just add a simple comment to the code that links to the first comment in this wee thread (I'm just a tad worried that this short-term solution becomes a longer-than-expected-term solution, so a comment in the code might at least catch someone's attention later if it does linger...!).